### PR TITLE
Update ML approach

### PR DESCRIPTION
**Pull request by amadeuspzs**
# Summary

**What/Why**

This PR updates the ML approach to move away from GridSearchCV (blank params) to:

1. Baseline models trained using default parameters and training set
2. Baseline models tested using test set
3. Best baseline model re-trained using GridSearchCV to produce final model
4. Final model tested using test set

**Your approach**

1. Implement a new "train_and_test_model" function in the helper file
2. Update both regression and classification notebooks to use the new function
3. Implement GridSearchCV once per notebook
4. Update other notebooks with relevant fixes

**What to review**

* src/utils.py
* 05a Modelling - Regression
* 06 Modelling - Classification
* 07 Model Comparison - note this could be a comparison of just the two final models, but kept all baseline models in now for minimum change 

# Checklists

This pull/merge request meets the following requirements:

- [ ] code runs
- [X] [developments are ethical][data-ethics-framework]
- [X] developments are secure (no data or notebook output)
- [X] you have made proportionate checks that the code works correctly
- [X] [minimum usable documentation][agilemodeling] written in the `docs` folder

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -28,49 +28,41 @@ def risk_score(los):
...
+
+    # fit the model based on the training data
+    model["model"] = estimator.fit(X_train, y_train)
+
+    # generate predictions of the training set
+    preds_train = np.clip(model["model"].predict(X_train), 0, None)
+    preds_test = np.clip(model["model"].predict(X_test), 0, None)
+
+    # calculate performance
+    if scoring_metric == "mean_squared_error":
```
Comment (matthewcooper19) [inline src/utils.py:59]: Could be useful to save the scoring metric into the model dictionary too, so that when the user picks up a set of models and prints the result they know the metric saved into the model when it was created..

File: notebooks/05a Modelling - regression.ipynb [hunk 2]
```diff
@@ -762,24 +924,80 @@
...
+    "# add statistical data in legend. LoA = limits of agreement\n",
+    "# flag: errors are not normally distibuted so does 2*std capture 95% interval?\n",
+    "\n",
+    "axs[1].legend(\n",
+    "    [\n",
+    "        f\"\\u03bc ({np.round(predictions_df.error.mean(),2)} days)\",\n",
+    "        f\"95% LoA (\\u03C3 {np.round(predictions_df.error.std(),2)} days gives {2*np.round(predictions_df.error.std(),2)})\",\n",
+    "    ]\n",
+    ")\n",
+    "axs[1].set_title(f\"{model_name} - MSE {mse.round(2)} days\")\n",
```
Comment (matthewcooper19) [inline notebooks/05a Modelling - regression.ipynb:998]: Should the units of MSE be days^2 rather than days here?

File: notebooks/05a Modelling - regression.ipynb [hunk 3]
```diff
@@ -686,18 +631,18 @@
...
     "    # inference - ensure smallest LoS is 0 days (not negative value)\n",
     "    preds = np.clip(models[model][\"model\"].predict(model_X_test), 0, None)\n",
     "\n",
-    "    # calculate MAE and range of LoS\n",
-    "    mae = mean_absolute_error(model_y_test, preds)\n",
+    "    # calculate MSE and range of LoS\n",
+    "    mse = mean_squared_error(model_y_test, preds)\n",
     "    print(\n",
-    "        f\"{model} test mae: {mae.round(3)} ({preds.min().round(1)} - {preds.max().round(1)} days)\"\n",
+    "        f\"{model} test mse: {mse.round(3)} ({preds.min().round(1)} - {preds.max().round(1)} days)\"\n",
```
Comment (matthewcooper19) [inline notebooks/05a Modelling - regression.ipynb:645]: Similar comment to above, is days the correct unit for MSE?

**Review by matthewcooper19 — CHANGES_REQUESTED** [review]
Wording around error units needs to be changed prior to merge

**Review by matthewcooper19 — APPROVED** [review]
